### PR TITLE
add Who's On First open data downloader

### LINKED ISSUE #16: Add downlaoder for Who's On First open admin data
Data is availabl here by ISO code: https://geocode.earth/data/whosonfirst/. 

The inventory JSON has full list: https://data.geocode.earth/wof/dist/shapefile/inventory.json

This pattern roughly matches the existing GADM downloader, to riff on that.

### PR DESCRIPTION
**Pull request by nvkelso**
Fixes https://github.com/thangqd/HCMGIS/issues/16.

Adds new downloader under:

> HCMGIS > Download Open Data > Global Administrative Areas by Country from WOF

Adapted from GADM downloader, but using Who's On First's particular modeling of country names and 2-char short codes. 

I am one of the primary maintainers for [Who's On First](https://whosonfirst.org/) gazetteer. We recently added [Shapefile downloads](https://whosonfirst.org/blog/2023/05/31/shapefiles/) and adding this option will make it easier to load the data into QGIS. This [blog post](https://whosonfirst.org/blog/2023/06/07/state-of-the-gazetteer/) covers WOF global data coverage.

### CODE REVIEW

File: .gitignore
Comment (nvkelso) [inline .gitignore:1]: I'm on a Mac, this hidden file gets created automatically. Adding it to .gitignore keeps repo clean for everyone else.

File: README.md
Comment (nvkelso) [inline README.md:12]: Here and in many other places my IDE removed trailing whitespace. No-op in terms of functionality, but makes the DIFF far more complicated. I can restore the whitespace if desired, just let me know!

File: icons/hcmgis_wof.png
Comment (nvkelso) [inline icons/hcmgis_wof.png:1]: This is the same image WOF uses as a favicon on our website, downsized. No rights issues.

File: metadata.txt
Comment (nvkelso) [inline metadata.txt:7]: The public releases don't follow a simple increment version approach? This may need to be updated to match what you have locally?

File: hcmgis_dialogs.py
Comment (nvkelso) [inline hcmgis_dialogs.py:1285]: Meat of new dialog logic for WOF

File: hcmgis_library.py
Comment (nvkelso) [inline hcmgis_library.py:2596]: Meat of new dialog logic for WOF

File: hcmgis_menu.py
Comment (nvkelso) [inline hcmgis_menu.py:400]: Meat of new dialog logic for WOF to add to plugin menu items

File: hcmgis_menu.py
Comment (nvkelso) [inline hcmgis_menu.py:558]: Meat of new dialog logic for WOF

**Comment by nvkelso** [comment]
@thangqd any chance for a review?

**Review by thangqd — APPROVED** [review]

**Comment by nvkelso** [comment]
@thangqd awesome, thanks! when do you expect to release into QGIS plugin registry? 🙏

**Comment by thangqd** [comment]
@nvkelso I've just uploaded to QGIS Plugin repo and waiting for approval. Thank you for your awesome contribution!

**Pull request merged**
